src/app/child-dev-project/notes/dashboard-widgets/no-recent-notes-dashboard/no-recent-notes-dashboard.component.ts

```diff
@@ -0,0 +1,66 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { ChildrenService } from '../../../children/children.service';
+import { Child } from '../../../children/model/child';
+import moment from 'moment';
+
+/**
+ * Dashboard Widget displaying children that do not have a recently added Note.
+ */
+@Component({
+  selector: 'app-no-recent-notes-dashboard',
+  templateUrl: './no-recent-notes-dashboard.component.html',
+  styleUrls: ['./no-recent-notes-dashboard.component.scss'],
+})
+export class NoRecentNotesDashboardComponent implements OnInit {
+  /**
+   * number of days since last note that children should be considered having a "recent" note.
+   *
+   * Defaults to beginning of the current week
+   */
+  @Input() sinceDays: number = moment().startOf('day').diff(moment().startOf('week'), 'days');
+
+  /** true while data is not ready/available yet */
+  isLoading: boolean;
+
+  /** children displayed in the template
+   * Child entities with additional "daysSinceLastNote" field
+   */
+  concernedChildren: ChildWithRecentNoteInfo[] = [];
+
+
+  constructor(
+    private childrenService: ChildrenService,
+  ) { }
+
+  async ngOnInit() {
+    await this.loadConcernedChildrenFromIndex();
+  }
+
+  private async loadConcernedChildrenFromIndex() {
+    this.isLoading = true;
+
+    const children = (await this.childrenService.getChildren().toPromise() as ChildWithRecentNoteInfo[])
+      .filter(c => c.isActive());
+
+    const lastNoteStats = await this.childrenService.getDaysSinceLastNoteOfEachChild();
+
+    const resultChildren = [];
+    for (const child of children) {
+      if (lastNoteStats.has(child.getId())) {
+        child.daysSinceLastNote = lastNoteStats.get(child.getId());
+        if (child.daysSinceLastNote > this.sinceDays) {
+          resultChildren.push(child);
+        }
+      } else {
+        child.daysSinceLastNote = 0;
```

Comment:
It seems reasonably to me to sort the list of concernedChildren descending by the number of days since the last note/report and it makes sense to have the children without any note/report at the end of the list. However displaying "0" for the number of days in the widget table could mean that the last note/report has been added or edited today. Maybe we could simply hide the "0" or display "no report" instead?

Reply:
I set it to 0 to allow a simple sort function. Updated the template now to improve the display for users. Good suggestion.